plugins/kubernetes/src/kubernetes-auth-provider/types.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import { AuthRequestBody } from '@backstage/plugin-kubernetes-backend';
+
+export interface KubernetesAuthProvider {
+  decorateRequestBodyForAuth(
+    requestBody: AuthRequestBody,
+  ): Promise<AuthRequestBody>;
+}
+
+export const KubernetesAuthProvidersApiRef = createApiRef<
```

Comment:
```suggestion
export const kubernetesAuthProvidersApiRef = createApiRef<
```

Reply:
Updated the latest commit with this change!